src/data/providers/initPoolDataProvider.ts

```diff
@@ -49,22 +50,39 @@ export class InitPoolDataProvider {
             address: poolAddress,
             publicClient: this.client,
         });
+
+        const poolTokens = amounts
+            .sort((a, b) => {
+                const diff = BigInt(a.address) - BigInt(b.address);
+                return diff > 0 ? 1 : diff < 0 ? -1 : 0;
+            })
+            .map(({ address, decimals }, index) => ({
+                address: address.toLowerCase() as Address,
+                decimals,
+                index: index + 1,
+            }));
+
+        const poolTokensWithBpt = sortTokensByAddress([
+            {
+                address: poolAddress.toLowerCase() as Address,
+                decimals: 18,
+                index: 0,
+            },
+            ...poolTokens,
+        ]);
```

Comment:
Since BPT is not necessarily at index 0, you should build/sort the tokens array only after adding the BPT, right?

Reply:
That's what's happening, it's sorting the array with the BPT token inside